src/docs/sphinx/dev_guide/release.rst

```diff
@@ -25,75 +25,111 @@ whenever, the master branch is changed, it is tagged with a new
+#. Make sure all documentation (source code, user guides, etc.) is
+   updated and reviewed. This should not be a substantial undertaking as
+   most of this should have been done during the regular development cycle.
 
-       $ git checkout master
-       $ git tag vMM.mm.pp
+#. Proofread the release notes for completeness and clarity and address
+   any shortcomings. Again, this should not take much time as release notes
+   should be updated during the regular development cycle. See
+   :ref:`release-notes-label` for information about release notes.
 
-     Recall that 'MM' is the major version number, 'mm' is the minor version
-     number, and 'pp' is the patch number. The command::
+3: Merge Release Candidate
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-       $ git tag
+Merge the release candidate branch into master branch once it is ready and
+approved. At this point, the release candidate branch can be deleted.
 
-     will list all the tags and you should see the new tag you created.
-     The command::
 
-       $ git show vMM.mm.pp
+4: Draft a Github Release
+^^^^^^^^^^^^^^^^^^^^^^^^^
 
-     will show information about the tagged commit.
+`Draft a new Release on Github <https://github.com/LLNL/axom/releases/new>`_
 
-  #. **Push the new tag to bitbucket.** You can push a single tag
-     to bitbucket by running the command ::
+#. Enter the desired tag version, e.g., v0.5.0
 
-       $ git push vMM.mm.pp
+#. Select **master** as the target branch to tag a release.
 
-  #. **Create a pull request to merge master into develop.** When approved,
-     merge it.
+#. Enter a Release title. We typically use titles of the following form *Axom-v0.3.1*
+
+#. Copy and paste the information for the release from the
+   ``axom/RELEASE-NOTES.md`` into the release description (omit any sections if empty).
+
+#. Publish the release. This will create a tag at the tip of the master
+   branch and add corresponding entry in the
+   `Releases section <https://github.com/LLNL/axom/releases>`_
+
+.. note::
+
+   Github will add a corresponding tarbal and zip archives consisting of the
+   source files for each release. However, these files do not include any
+   submodules. Consequently, a tarball that includes all of the submodules is
+   generated manually in a seperate step.
+
+5: Make a Release Tarball
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Checkout the master branch locally and run the ``axom/scripts/make_release_tarball.sh``
+  script. This will generate a tarball of the form ``Axom-v0.3.1.tar.gz``
+
+* Upload the tarball for the corresponding release, by going to the
+ `Github Releases section <https://github.com/LLNL/axom/releases>`_ and ``Edit``
+ the release created earlier.
+
+* Attach the tarball to the release.
+
+* Add a note at the top of the release description that indicates which
+  tarbal consists of all the submodules, e.g., *\"Please download the Axom-v0.3.1.tar.gz tarball below, which includes all of the Axom submodules as well\"*
```

Comment:
tarbal -> tarball

Reply:
> Add a note at the top of the release description that indicates which
  tarbal consists of all the submodules

Consider changing this to something like:

> Add a note at the top of the release description to highlight that the tarball includes of all the submodules.

Reply:
Fixed the typo. There are two tarballs: (a) the one generated by Github and (b) the one generated manually and uploaded. So the comment needs to indicate which of the two tarballs also includes the submodules.